gnd/src/main/java/com/google/android/gnd/repository/OfflineAreaRepository.java

```diff
@@ -170,13 +180,28 @@ public Completable addAreaAndEnqueue(LatLngBounds bounds) {
         .flatMapPublisher(
             tiles ->
                 getDownloadedTileSourcesOnceAndStream()
-                    .map(ts -> stream(ts).filter(tiles::contains).collect(toImmutableSet())))
+                    .map(
```

Comment:
Any way we can reduce nesting here? Maybe just refactor part of this into its own method?

Reply:
b86a87a